### PR TITLE
Clean up ModulesModelModule class

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This PR make some small changes to ModulesModelModule class:

1. Use local $db variable instead of $this->_db
2. Use JDatabaseQuery to insert module assignment instead of using plain SQL
3. Re-use $query object instead of creating new instance
4. Remove one time usage $app variable, call JFactory::getApplication()->redirect directly

### Testing Instructions
1. Code review
2. Access to Extensions -> Modules, click on a module to edit, try to change Module Assignment, make sure it is saved properly.
### Documentation Changes Required
None

### CODE REVIEW

File: administrator/components/com_modules/models/module.php [hunk 1]
```diff
@@ -426,8 +426,8 @@ public function duplicate(&$pks)
 					->from($db->quoteName('#__modules_menu'))
 					->where($db->quoteName('moduleid') . ' = ' . (int) $pk);
 
-				$this->_db->setQuery($query);
-				$rows = $this->_db->loadColumn();
+				$db->setQuery($query);
+				$rows = $db->loadColumn();
```
Comment (andrepereiradasilva) [inline administrator/components/com_modules/models/module.php]: for further cleanup i think you could use just
```php
$rows = $db->setQuery($query)->loadColumn();
```

File: administrator/components/com_modules/models/module.php [hunk 2]
```diff
@@ -448,11 +448,11 @@ public function duplicate(&$pks)
 				->columns($db->quoteName(array('moduleid', 'menuid')))
 				->values($tuples);
 
-			$this->_db->setQuery($query);
+			$db->setQuery($query);
 
 			try
 			{
-				$this->_db->execute();
+				$db->execute();
```
Comment (andrepereiradasilva) [inline administrator/components/com_modules/models/module.php]: same as above
```php
$db->setQuery($query)->execute();
```

File: administrator/components/com_modules/models/module.php (hunk 1 above)
Comment (joomdonation) [inline administrator/components/com_modules/models/module.php]: Thanks for your suggestion @andrepereiradasilva . The suggested change will works OK, however, seems we don't have that rule in core code. You can see on the same file (and many other places in Joomla core), we usually call $db->setQuery method to set query object before running the next method

With that said, I will leave it as how it is for now unless maintainers request for change.

Comment (andrepereiradasilva) [inline administrator/components/com_modules/models/module.php]: sure was just a suggestion.

Comment (joomdonation) [inline administrator/components/com_modules/models/module.php]: Since you reviewed the code, could you please help marking the test result, too?

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 9d65eb41590e0064fd7f153e9c3dd25dc426738c<br /><br />Code review OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13380">issues.joomla.org/tracker/joomla-cms/13380</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 9d65eb41590e0064fd7f153e9c3dd25dc426738c<br /><br />Code review OK and it works OK assigning modules to menu items.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13380">issues.joomla.org/tracker/joomla-cms/13380</a>.</sub>

**Comment by joomdonation** [comment]
@andrepereiradasilva When you have a chance please report your test result for this PR. It is nice to have this abit better code merged.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 9d65eb41590e0064fd7f153e9c3dd25dc426738c<br /><br />Test without Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13380">issues.joomla.org/tracker/joomla-cms/13380</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13380">issues.joomla.org/tracker/joomla-cms/13380</a>.</sub>

**Pull request merged**